webrender/src/internal_types.rs

```diff
@@ -342,10 +342,18 @@ pub enum RenderTargetMode {
 }
 
 pub enum TextureUpdateOp {
+    // Create(image.width, image.height, format, filter, mode, buffer)
     Create(u32, u32, ImageFormat, TextureFilter, RenderTargetMode, Option<Arc<Vec<u8>>>),
+    // Create(image.width, image.height, format, filter, mode, external_image_id)
+    CreateForExternalBuffer(u32, u32, ImageFormat, TextureFilter, RenderTargetMode, ExternalImageId),
+    // Update(page_pos.x, page_pos.y, image_size.width, image_size.height, buffer, stride)
     Update(u32, u32, u32, u32, Arc<Vec<u8>>, Option<u32>),
+    // UpdateExternalBuffer(alloc.x, alloc.y, alloc.width, alloc.height,
+    //                      request.x, request.y, request.width, request.height,
+    //                      external_image_id, bpp, stride)
+    UpdateForExternalBuffer(u32, u32, u32, u32, u32, u32, u32, u32, ExternalImageId, u32, Option<u32>),
```

Comment:
If we try to find the texture info at renderer, we should reference to the texture structure in device[[1]](https://github.com/JerryShih/webrender/blob/23123c85ddfccf4c458b9c3856890869644be667/webrender/src/device.rs#L804). Then, we need to have additional accessor for this texture structure and a bpp calculation function in render or device. Just pass a bpp is the simplest way.

[1]
https://github.com/JerryShih/webrender/blob/23123c85ddfccf4c458b9c3856890869644be667/webrender/src/device.rs#L804